server/models/estimationFeatureModel.js

```diff
@@ -126,5 +126,79 @@ estimationFeatureSchema.statics.addFeatureByEstimator = async (featureInput, est
+    if (!estimation)
+        throw new AppError('Estimation not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    if (!_.includes([SC.STATUS_ESTIMATION_REQUESTED, SC.STATUS_CHANGE_REQUESTED], estimation.status))
+        throw new AppError("Estimation has status as ["+estimation.status+"]. Estimator can only update feature into those estimations where status is in [" + SC.STATUS_ESTIMATION_REQUESTED + ", " + SC.STATUS_CHANGE_REQUESTED + "]", EC.INVALID_OPERATION, EC.HTTP_BAD_REQUEST)
+
+    if (estimationFeature.repo && estimationFeature.repo._id) {
+        featureInput.repo = {
+            _id: estimationFeature.repo._id
+        }
+        let updateFeatureFromRepo = await RepositoryModel.updateFeature(featureInput,estimator)
+    }
+
+    estimationFeature.technologies = featureInput.technologies
+    estimationFeature.tags = featureInput.tags
+
+    if (!_.isEmpty(featureInput.notes)) {
+        featureInput.notes = featureInput.notes.map(n => {
+            n.name = estimator.fullName
+            return n
+        })
+    }
+    let  mergeAllNotes = estimationFeature.notes
+    if (!_.isEmpty(mergeAllNotes)) {
+       featureInput.notes.map(n => {
+            mergeAllNotes.push(n)
+        })
+    }else{
+        mergeAllNotes = featureInput.notes
+    }
+    estimationFeature.notes = mergeAllNotes
+    estimationFeature.updated = Date.now()
+
+    return await estimationFeature.save()
+
+}
+
+estimationFeatureSchema.statics.moveToFeatureByEstimator = async (featureInput, estimator) => {
+    validate(featureInput, estimationEstimatorMoveToFeatureStruct)
+
+    if (!estimator || !userHasRole(estimator, SC.ROLE_ESTIMATOR))
+        throw new AppError('Not an estimator', EC.INVALID_USER, EC.HTTP_BAD_REQUEST)
+
+    let estimationFeature = await EstimationFeatureModel.findById(featureInput.feature_id)
+    if(!estimationFeature)
+        throw new AppError('Estimation feature not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    let estimationTask = await EstimationTaskModel.findById(featureInput.task_id)
+    if(!estimationTask)
+        throw new AppError('Estimation task not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    let estimation = await EstimationModel.findById(estimationFeature.estimation._id)
+    if (!estimation)
+        throw new AppError('Estimation not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    if (!_.includes([SC.STATUS_ESTIMATION_REQUESTED, SC.STATUS_CHANGE_REQUESTED], estimation.status))
+        throw new AppError("Estimation has status as ["+estimation.status+"]. Estimator can only update feature(Move to Feature) into those estimations where status is in [" + SC.STATUS_ESTIMATION_REQUESTED + ", " + SC.STATUS_CHANGE_REQUESTED + "]", EC.INVALID_OPERATION, EC.HTTP_BAD_REQUEST)
+
+    let taskOfEstimation = await EstimationTaskModel.updateTaskMoveToFeatureOfEstimation(featureInput.task_id,featureInput.feature_id,estimation._id)
```

Comment:
As this method would be moved to estimationTaskModel, merge code written here with code of updateTaskMoveToFeatureOfEstimation method. So there should be a single method for this